### PR TITLE
[4.2] Improve IE detection warning

### PR DESCRIPTION
**Pull request by brianteeman**
Currently if you try to login to the admin using internet explorer you will see a warning message.

image

This message is hidden from all other browsers with css.

The recent changes with child templates etc resulted in some users loading there site without any css. As a result they saw the warning which of course was not helpful as they were not using IE.
image

This PR extends the test for ie by using the api to absolutely ensure the message will only be displayed in the targeted browsers.

Testing is quite easy
1. Using IE try and log in to the admin and you will see the warning message both with and without this pr

image

2. With any other browser view the source before this PR you will see the following alert.

image

3. With any other browser view the source after this PR and there is no alert present

image

This resolves numerous duplicate issue reports but I think they have all been closed.

### CODE REVIEW

**Comment by HLeithner** [comment]
Maybe we can remove it completely? IE is finally finally finally (with exception of course) EOL on 15.06.2022

**Comment by brianteeman** [comment]
> Maybe we can remove it completely? IE is finally finally finally (with exception of course) EOL on 15.06.2022

perhaps but then if it does no harm having it and it helps some people then why not.

**Comment by HLeithner** [comment]
Because keeping legacy code when not needed should be removed because maintaining costs time. But yes could be done later too.

**Comment by brianteeman** [comment]
Until IE usage decreases to much lower I would still like to keep it.

This PR is just making it work as intended

File: administrator/templates/atum/login.php [hunk 1]
```diff
@@ -10,6 +10,7 @@
 defined('_JEXEC') or die;
 
 use Joomla\CMS\Factory;
+use Joomla\CMS\Environment\Browser;
```
Comment (heelc29) [inline administrator/templates/atum/login.php]:
```suggestion
use Joomla\CMS\Environment\Browser;
use Joomla\CMS\Factory;
```

Comment (brianteeman) [reply administrator/templates/atum/login.php]: oops - thanks

**Comment by brianteeman** [comment]
rebased to 4,2 now that there will be no further 4.1 releases

**Comment by brianteeman** [comment]
This super simple pr would save so many mistaken error reports on the forum

**Comment by N6REJ** [comment]
@brianteeman i don't know the relevance since I can't recreate it but I opened an articled directly in edge accidently, meant to click on firefox, and the error you showed came up.  But when I tried it specifically to test this pr it opened fine without any notices or issues.

**Comment by brianteeman** [comment]
Edge != IE

**Comment by brianteeman** [comment]
Every day there are more reports of people being confused by seeing this error message when they are not using ie. Just test and merge this and they go away

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on bb3066dcdd0d12c3e16faa3106e2dfb20bf53df3<br /><br />code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37900">issues.joomla.org/tracker/joomla-cms/37900</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on bb3066dcdd0d12c3e16faa3106e2dfb20bf53df3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37900">issues.joomla.org/tracker/joomla-cms/37900</a>.</sub>

**Comment by N6REJ** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37900">issues.joomla.org/tracker/joomla-cms/37900</a>.</sub>

**Comment by bembelimen** [comment]
CMS maintenance decided to remove the message following @HLeithner recommendation.

Thanks for the initial pointer.

**Comment by brianteeman** [comment]
Nothing has changed in our code to justify removing the error message. Joomla admin still will not work with internet explorer but with this PR no one will know that the browser is the cause of the problems and will now think Joomla is broken.

Internet Explorer is still in use even if it is eol and a user should be told when they get an error.

Just because in the wealthy western web the usage of IE has dropped below 1% does not mean that it is true globally.

For example in Iran, where joomla is heavily used due to our almost unique, support for the jihali calendar ie usage is still over 8% https://gs.statcounter.com/browser-market-share/desktop/iran

There are many other countries where Joomla is actively used where the % is 5% or over.

**Comment by brianteeman** [comment]
Please re-open as #39018 has been moved to 5.0

**Comment by zero-24** [comment]
Done as requested

**Comment by brianteeman** [comment]
Will this now be merged.

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**